lightning/src/ln/channel.rs

```diff
@@ -5119,8 +5119,46 @@ impl<Signer: WriteableEcdsaChannelSigner> Channel<Signer> {
 						return Ok((Some(channel_ready), announcement_sigs));
 					}
 				}
+
+				// If we have a vanilla LN channel, this checks if the transaction
+				// spends from the actual funding output.
+				//
+				// If we have a split channel, this checks if the transaction spends
+				// from the glue output.
+				//
+				// In both cases we are checking if this transaction is the
+				// commitment transaction.
+				let is_funding_or_glue_txo = |prev_outpoint: &bitcoin::OutPoint| -> bool {
+					prev_outpoint == &funding_txo.into_bitcoin_outpoint()
+				};
+
+				// This check only runs if the check above returns `false`.
+				//
+				// Given that, if we have a vanilla LN channel this check will
+				// return `false` too, because for vanilla LN channels `funding_txo
+				// == original_funding_txo`.
+				//
+				// If we have a split channel, this checks if the transaction spends
+				// from the actual funding output BUT it is _not_ the split
+				// transaction. This can happen if the split channel is closed using
+				// a revoked commitment transaction!
+				//
+				// We purposefully do not announce the closure of the channel with
+				// the confirmation of the split transaction. TODO(lucas): Exactly
+				// why is this, Tibo?
```

Comment:
Because AFAIU LDK assumes that if a channel is closed by something else than a commitment transaction then it is a mutual close, and will then not react on the actual commitment transaction that comes after (and not trigger the event to claim the funds from it).